signac/synced_collections/data_types/synced_attr_dict.py

```diff
@@ -186,7 +186,10 @@ def __setitem__(self, key, value):
         # should not have side effects once that backwards compatibility layer
         # is removed, so we can validate a temporary dict {key: value} and
         # directly set using those rather than looping over data.
+
         data = {key: value}
+        # TODO: May need to convert from base prior to validation to handle
```

Comment:
No, the implemented validators themselves handle this correctly. I think it makes sense to do this beforehand because validators should have the flexibility to respond to special types in different ways before they are coerced, if that is necessary.